### PR TITLE
[Core] Attempt at fixing method overload resolving for MSBuild

### PR DESCRIPTION
**Pull request by Therzok**
Fixes VSTS #783960 - MSBuild property evaluation failed

### CODE REVIEW

**Comment by mrward** [comment]
Needs rebasing to fix the build error.

**Comment by Therzok** [comment]
Failures seem unrelated

**Review by mrward — COMMENTED** [review]
Tested this locally by opening Main.sln and see the following logged in the IDE log many times:

```
MSBuild property evaluation failed: Registry:HKEY_LOCAL_MACHINE\Software\Microsoft\.NETFramework@InstallRoot
System.IndexOutOfRangeException: Index was outside the bounds of the array.
  at MonoDevelop.Projects.MSBuild.MSBuildEvaluationContext.EvaluateMember (System.Type type, System.Object instance, System.ReadOnlySpan`1[T] str, System.Int32 i, System.Object& val) [0x000c4] in /Users/vsts/agent/2.155.1/work/1/s/monodevelop/main/src/core/MonoDevelop.Core/MonoDevelop.Projects.MSBuild/MSBuildEvaluationContext.cs:638 
```

The old error about the System.String.Substring/LastIndexOf failing to be evaluated is fixed.

File: main/src/core/MonoDevelop.Core/MonoDevelop.Projects.MSBuild/MSBuildEvaluationContext.cs [hunk 1]
```diff
@@ -634,6 +634,7 @@ internal bool EvaluateMember (Type type, object instance, ReadOnlySpan<char> str
 			} else {
 				// It has to be a property or field
 				try {
+					var member = ResolveMember (type, memberName.ToString (), instance == null, MemberTypes.Property | MemberTypes.Field);
```
Comment (Therzok) [inline main/src/core/MonoDevelop.Core/MonoDevelop.Projects.MSBuild/MSBuildEvaluationContext.cs]: This needed a null/empty check

**Review by slluis — APPROVED** [review]

**Comment by Therzok** [comment]
Verified latest version, no exception in the logs.

**Review by mrward — APPROVED** [review]

**Pull request merged**